File: kislate_workshop/luogu_A-Z_2025/chooseW.py
import math

def is_prime(n):
    if n == 1:
        return False
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True

def countSum_Prime(array, n, k, ans, sum, index):
    if k == 0:
        if is_prime(sum):
            ans[0] += 1
        return
    for i in range(index, n):
        countSum_Prime(array, n, k - 1, ans, sum + array[i], i + 1)

n, k = map(int, input().split())
array = [int(x) for x in input().split()]
ans = [0]
countSum_Prime(array, n, k, ans, 0, 0)
print(ans[0])

File: kislate_workshop/Problems/luogu/winter_2025/luogu_A-Z_2025/chooseW.cpp
/*
选择（choose）
【题目描述】
给你 n 个数和一个整数 k。
问有你有多少种方法从 n 个数里面选 k 个数使得和为质数？
【输入格式】
第一行两个空格隔开的整数 n, k。
第二行 n 个整数，分别为 x1, x2, ···, xn。
【输出格式】
输出一个整数，表示种类数。
【样例 1 输入】
4 3
3 7 12 19
【样例 1 输出】
1
【子任务】
数据保证：1 ≤n ≤20，k < n，1 ≤xi ≤5 ×106。
*/
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>
using namespace std;
bool isPrime(int n){
    if(n==1)
        return false;
    for(int i = 2; i < sqrt(n); i++){
        if(n%i == 0)
            return false;
    }
    return true;
}

void countCombinationPrime(vector<int>&array, int n, int k, int& ret, int sum, int index){
    if(k == 0){
        if(isPrime(sum))
            ret++;
        return;
    }
    for(int i = index; i < n; i++){
        countCombinationPrime(array, n, k - 1, ret, sum + array[i], i + 1);
    }
}
int main(void){
    int n, k;
    cin >> n >> k;
    vector<int> array(n);
    for(int i = 0; i < n; i++){
        cin >> array[i];
    }
    int ret = 0;
    countCombinationPrime(array, n, k, ret, 0, 0);
    printf("%d\n", ret);
    return 0;
}